### DISCUSSION TITLE
Unit Testing

### DISCUSSION THREAD

**Discussion by khalidsaadat** [khalidsaadat/soen341#123] [Announcements]
Hi everyone! I wanted to ask if we should go with PHPUnit or PHPMD for unit testing? any suggestion?

**Comment by shabiasaeed** [comment]
I would suggest PHPMD as it was also recommended by the professor.

**Reply by khalidsaadat** [reply]
After checking around, I found that PHPMD is for mess detection, while PHPUnit is for unit testing. So I would suggest we go with PHPUnit as it is similar to jUnit

**Comment by khalidsaadat** [comment]
I found a solution for PHPUnit using composer. I will show how it works during our meeting tonight.

**Comment by AnumSidd** [comment]
Yes. PHPUnit would be good.
